src/gc/unix/cgroup.cpp

```diff
@@ -59,42 +63,44 @@ class CGroup
 
     static bool GetPhysicalMemoryLimit(uint64_t *val)
     {
+        const char *files[] = {
+            CGROUP1_MEMORY_LIMIT_FILENAME,
+            CGROUP2_MEMORY_LIMIT_FILENAME
+        };
+
         char *mem_limit_filename = nullptr;
         bool result = false;
 
         if (s_memory_cgroup_path == nullptr)
             return result;
 
-        size_t len = strlen(s_memory_cgroup_path);
-        len += strlen(MEM_LIMIT_FILENAME);
-        mem_limit_filename = (char*)malloc(len+1);
+        mem_limit_filename = SearchForFile(s_memory_cgroup_path, files, sizeof(files));
```

Comment:
`sizeof(array)` doesn't do what I thought it would do. This should be `_countof()` or something similar.